lib/galaxy/jobs/__init__.py

```diff
@@ -1851,9 +1851,10 @@ def __update_output(self, job, hda, clean_only=False):
                 # to the object store from jobs - be sure that file is cleaned up. This
                 # is a bit of hack - our object store abstractions would be stronger
                 # and more consistent if tools weren't writing there directly.
-                target = dataset.file_name
-                if os.path.exists(target):
-                    os.remove(target)
+                try:
```

Comment:
xref https://github.com/galaxyproject/galaxy/commit/009e2cbf5e9592fd8af782b2cde36d8d1687c003

Reply:
Not sure I'm seeing your reasoning there for why you're not using `.delete()`?

Reply:
That was for my own tracking purposes, I'm sure what you have right now is better.